app/controllers/api/v1/users_controller.rb

```diff
@@ -14,19 +15,12 @@ def show
     render json: @users
   end
 
-  # The create method is handled by Devise users/registration
-  # def create
-  #   @user = User.new(user_params)
-  #   if @user.save
-  #     render json: { message: "User created successfully!" }, status: :created
-  #   else
-  #     render json: { errors: @user.errors.full_messages }, status: :unprocessable_entity
-  #   end
-  # end
-
   # PATCH/PUT api/v1/users/1
   def update
-    if @user.update(user_params)
+    @user = User.find(params[:id])
+    if params[:user].key?(:role)
+      render json: { error: "You are not authorized to change the role" }, status: :forbidden
```

Comment:
You don't need this condition.

Instead, add a [`rescue_from`](https://apidock.com/rails/ActiveSupport/Rescuable/ClassMethods/rescue_from) to your `ApplicationController` to respond with `status: :forbidden` when `ActionController::UnpermittedParameters` is raised. This is much DRY-er.

Then, when you say `if @user.update(user_params)`, it will invoke `ActionController::Parameters`, and will raise that error if necessary.

Reply:
done.